src/twisted/plugins/twisted_reactors.py

```diff
@@ -4,59 +4,70 @@
 from __future__ import absolute_import, division
 
 from twisted.application.reactors import Reactor
-from twisted.python.compat import _PY3
+from twisted.python.reflect import requireModule
 
-default = Reactor(
+def _addReactorToAll(shortName, moduleName, description):
+    """
+
+    @param shortName: Short name of reactor
+    @param moduleName: The fully-qualified module name of the reactor
+    @param description: Description of reactor
+    @return: newly created reactor
+    @rtype: L{Reactor} or L{None}
+    """
+    newReactor = None
+    if requireModule(moduleName) is not None:
```

Comment:
This strikes me as a recipe for silent failure.  Some reactors we expect to work in certain configurations; other reactors we expect to be absent.  I think that trying to dynamically figure this out without recording the error anywhere might be confusing.